### PR TITLE
Remove contracts + Adapt for new contracts

### PR DESCRIPTION
**Pull request by woksin**
* Removes Contracts project
* Removes Module
* Changes ExecutionContext data
* Extension methods and utility for new contracts

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1140654762299587/1171851369249975)

### CODE REVIEW

File: Source/Protobuf/Extensions.cs [hunk 1]
```diff
@@ -16,113 +18,94 @@ namespace Dolittle.Protobuf
...
-        /// <returns>Converted <see cref="ConceptAs{T}"/> of type <see cref="System.Guid"/>.</returns>
-        public static T To<T>(this ByteString idAsBytes)
-            where T : ConceptAs<System.Guid>, new()
-        {
-            return new T { Value = new System.Guid(idAsBytes.ToByteArray()) };
-        }
+        /// <param name="id"><see cref="grpc.Uuid"/> to convert.</param>
+        /// <returns>Converted <see cref="ConceptAs{T}"/> of type <see cref="Guid"/>.</returns>
+        public static T To<T>(this grpc.Uuid id)
+            where T : ConceptAs<System.Guid>, new() => new T { Value = new Guid(id.Value.ToByteArray()) };
```
Comment (jakhog) [inline Source/Protobuf/Extensions.cs:30]: Reuse logic.
```suggestion
        public static T To<T>(this grpc.Uuid id)
            where T : ConceptAs<System.Guid>, new() => new T { Value = id.ToGuid() };
```

File: Source/Protobuf/Extensions.cs [hunk 2]
```diff
@@ -16,113 +18,94 @@ namespace Dolittle.Protobuf
...
-        /// <param name="id"><see cref="ConceptAs{T}"/> of type <see cref="System.Guid"/> to convert.</param>
-        /// <returns>Converted <see cref="ByteString"/>.</returns>
-        public static ByteString ToProtobuf(this ConceptAs<System.Guid> id)
-        {
-            return ByteString.CopyFrom(id.Value.ToByteArray());
-        }
+        /// <param name="id"><see cref="ConceptAs{T}"/> of type <see cref="Guid"/> to convert.</param>
+        /// <returns>Converted <see cref="grpc.Uuid"/>.</returns>
+        public static grpc.Uuid ToProtobuf(this ConceptAs<Guid> id) =>
+            new grpc.Uuid { Value = ByteString.CopyFrom(id.Value.ToByteArray()) };
```
Comment (jakhog) [inline Source/Protobuf/Extensions.cs:53]: Reuse logic.
```suggestion
        public static grpc.Uuid ToProtobuf(this ConceptAs<Guid> id) => id.Value.ToProtobuf();
```

File: Source/Protobuf/Extensions.cs [hunk 3]
```diff
@@ -16,113 +18,94 @@ namespace Dolittle.Protobuf
...
-        /// <returns>Converted <see cref="Claims"/>.</returns>
-        public static Claims ToClaims(this IEnumerable<Security.Contracts.Claim> claims)
-        {
-            return new Claims(claims.Select(_ => new Claim(_.Key, _.Value, _.ValueType)));
-        }
+        /// <param name="failureReason">The reason for failure.</param>
+        /// <param name="failureId">The failure id.</param>
+        /// <returns>Converted <see cref="grpc.Failure" />.</returns>
+        public static grpc.Failure ToProtobufFailure(this string failureReason, Guid failureId) =>
+            new grpc.Failure { Reason = failureReason, Id = failureId.ToProtobuf() };
```
Comment (jakhog) [inline Source/Protobuf/Extensions.cs:84]: I don't think we should have these - we want to always construct from the `Failure`type.

I can't add a suggestion to remove, but my suggestion is just to get rid of them.

File: Source/Protobuf/Extensions.cs [hunk 4]
```diff
@@ -16,113 +18,94 @@ namespace Dolittle.Protobuf
...
         /// Convert to metadata that is used as arguments in the header.
         /// </summary>
         /// <param name="message"><see cref="IMessage"/> to convert.</param>
         /// <returns>A <see cref="Metadata.Entry"/> that can be used directly.</returns>
-        public static Metadata.Entry ToArgumentsMetadata(this IMessage message)
-        {
-            return new Metadata.Entry(_argumentsKey, message.ToByteArray());
-        }
+        public static Metadata.Entry ToArgumentsMetadata(this IMessage message) =>
+            new Metadata.Entry(_argumentsKey, message.ToByteArray());
```
Comment (jakhog) [inline Source/Protobuf/Extensions.cs:108]: We shouldn't use this anymore - so get rid of it as well.

File: Source/Protobuf/Extensions.cs [hunk 5]
```diff
@@ -16,113 +18,94 @@ namespace Dolittle.Protobuf
     /// </summary>
     public static class Extensions
     {
-        static string _argumentsKey = $"arguments{Metadata.BinaryHeaderSuffix}";
+        static readonly string _argumentsKey = $"arguments{Metadata.BinaryHeaderSuffix}";
```
Comment (jakhog) [inline Source/Protobuf/Extensions.cs:21]: We shouldn't use this anymore.
```suggestion
```

File: Source/Protobuf/Extensions.cs (hunk 3 above)
Comment (woksin) [reply Source/Protobuf/Extensions.cs]: Yes, they were removed in a later PR. But I removed them here :)

File: Source/Protobuf/Extensions.cs (hunk 4 above)
Comment (woksin) [reply Source/Protobuf/Extensions.cs]: 👍

File: Specifications/Protobuf/for_Extensions/when_converting_execution_context_to_protobuf.cs [hunk 6]
```diff
@@ -21,7 +21,7 @@ public class when_converting_execution_context_to_protobuf
             execution_context = new ExecutionContext(
                 Guid.NewGuid(),
                 Guid.NewGuid(),
-                Guid.NewGuid(),
+                Versioning.Version.NotSet,
```
Comment (jakhog) [inline Specifications/Protobuf/for_Extensions/when_converting_execution_context_to_protobuf.cs:24]: Perhaps it's better to test with an actual Version? Just worried that `NotSet` is a default value somewhere so that we won't see if it is actually passed through correctly.

File: DotNET.Fundamentals.sln [hunk 7]
```diff
@@ -77,8 +77,6 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Rules", "Source\Rules\Rules
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "ResourceTypes", "Source\ResourceTypes\ResourceTypes.csproj", "{6FACBDA5-1C3D-43AE-BEAE-FD39EDC6A4FD}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "ResourceTypes", "Specifications\ResourceTypes\ResourceTypes.csproj", "{77270B8D-52DF-4A95-BE22-B2CE234E2657}"
```
Comment (jakhog) [inline DotNET.Fundamentals.sln:80]: Why was this removed? Seems unrelated to what we are doing...

File: Specifications/Protobuf/for_Extensions/when_converting_execution_context_to_protobuf.cs (hunk 6 above)
Comment (woksin) [reply Specifications/Protobuf/for_Extensions/when_converting_execution_context_to_protobuf.cs:24]: Agree

File: DotNET.Fundamentals.sln (hunk 7 above)
Comment (woksin) [reply DotNET.Fundamentals.sln:80]: It is. I was just in a cleanup mood and removed lots of folders on my local system
